src/main/java/com/avispl/symphony/dal/infrastructure/management/philips/huebridge/PhilipsHueDeviceCommunicator.java

```diff
@@ -3615,9 +3826,22 @@ private AutomationResponse convertAutomationByValues(String property, Map<String
 			timeAndRepeat.setDays(dayArrays);
 			TimePoint timePoint = new TimePoint();
 			CurrentTime currentTime = new CurrentTime();
+			if (PhilipsConstant.CREATE_AUTOMATION.equalsIgnoreCase(property)) {
+				timeCurrent = timeAndMinuteForCreateAutomation.get(typeOfAutomation).get(AutomationEnum.TIME_CURRENT.getName());
+				timeHour = timeAndMinuteForCreateAutomation.get(typeOfAutomation).get(AutomationEnum.TIME_HOUR.getName());
+				timeMinute = timeAndMinuteForCreateAutomation.get(typeOfAutomation).get(AutomationEnum.TIME_MINUTE.getName());
+			}
 			if (String.valueOf(PhilipsConstant.NUMBER_ONE).equals(timeCurrent)) {
-				int time = Integer.parseInt(timeHour) + 12;
+				int time = Integer.parseInt(timeHour);
 				timeHour = String.valueOf(time);
+				if (time != 12) {
+					timeHour = String.valueOf(time + 12);
```

Comment:
can you please explain the logic here? So the time returned in 12 hour format and you add 12 to make it 24 format?
Because later on, in the next if-else section it makes 12 to be 0, which would happen twice for 12-hour time format... Maybe I'm missing something... TimeCurrent variable is defining the am/pm period, or is it something else?